solidity/contracts/factory.sol

```diff
@@ -20,54 +20,67 @@ import {IZetoFungibleInitializable} from "./lib/interfaces/zeto_fungible_initial
 import {IZetoNonFungibleInitializable} from "./lib/interfaces/zeto_nf_initializable.sol";
 
 contract ZetoTokenFactory {
+    struct InitArgs {
+        address implementation;
+        address depositVerifier;
+        address withdrawVerifier;
+        address verifier;
+    }
+
     event ZetoTokenDeployed(address indexed zetoToken);
 
-    mapping(string => address) internal implementations;
+    mapping(string => InitArgs) internal implementations;
 
     constructor() {}
 
     function registerImplementation(
         string memory name,
-        address implementation
+        InitArgs memory implementation
     ) public {
         implementations[name] = implementation;
     }
```

Comment:
This nesting of implementation address in a struct feels a bit weird

Reply:
agreed. I think that's mainly due to the naming. changing it from `InitArgs` to `ImplementationInfo` with some inline comments should make it feel more natural.